todos/sagas.js

```diff
@@ -0,0 +1,47 @@
+import { takeEvery } from 'redux-saga'
+import { call, put } from 'redux-saga/effects'
+import * as request from 'superagent'
+
+import { actionTypes as actions } from './constants'
+
+function * getTodos (action) {
+  try {
+    yield put({ type: actions.FETCHING })
+    const { todos } = yield call(getRequest)
+    yield put({ type: actions.FETCH_SUCCESS, todos })
+  } catch (error) {
+    yield put({ type: actions.FETCH_ERROR, error })
+  }
+}
+
+// Fetch todos whenever receive a FETCH action
+function * watchTodos () {
+  console.log(actions.FETCH)
+
+  yield * takeEvery(actions.FETCH, getTodos)
+}
+
+const getRequest = () => new Promise((resolve, reject) => {
```

Comment:
Yep - also want to move to superagent - get the backend up - get some nicely mocked tests etc.